server/routes/api/chart.py

```diff
@@ -110,6 +126,70 @@ def get_landing_page_data(dcid):
     )
     return response
 
+def get_date_to_add_for_chart(chart, sv_data):
+    """Get the date for which data is needed for a specific chart
+    
+    Args:
+        chart: the chart object that we currently care about (a single chart object from chart_config.json)
+        sv_data: the object returned from get_landing_page_data
+    
+    Returns:
+        date as a string
+    """
+    stat_vars = []
+    dates = set()
+    dates_initialized = False
+    for sv in chart['statsVars']:
+        if sv in sv_data:
+            if dates_initialized:
+                dates = dates.intersection(sv_data[sv]['data'])
+            else:
+                dates.update(sv_data[sv]['data'])
+        dates_initialized = True
+    sorted_dates = sorted(dates)
+    date_to_add = ''
+    if len(sorted_dates) > 0:
+        date_to_add = sorted_dates[-1]
+    return date_to_add
+    
+    
+def get_dates_for_stat_vars(chart_config, sv_data):
+    """Get the dates needed for each stat var
```

Comment:
could you add more here as well? mention that this checks all charts for the stat var.